src/txn/txn_ckpt.c

```diff
@@ -580,11 +582,30 @@ __checkpoint_prepare(WT_SESSION_IMPL *session, const char *cfg[])
 	txn_global = &conn->txn_global;
 	txn_state = WT_SESSION_TXN_STATE(session);
 
+	/*
+	 * Someone giving us a specific timestamp overrides the general
+	 * use_timestamp.
+	 */
 	WT_RET(__wt_config_gets(session, cfg, "read_timestamp", &cval));
 	if (cval.len > 0) {
 		WT_RET(__wt_snprintf(timestamp_config, sizeof(timestamp_config),
 		    "read_timestamp=%.*s", (int)cval.len, cval.str));
 		txn_cfg[2] = timestamp_config;
+	} else if (txn_global->has_oldest_timestamp) {
+		WT_RET(__wt_config_gets(session, cfg, "use_timestamp", &cval));
```

Comment:
While not directly related to your question, this code has a bug, in that it should be using `has_stable_timestamp` now.  And you're right about needing to figure out what that use case should be where the user asks for a timestamp and none has been set.  At a minimum here I'll document what happens on a mismatch.